From f2d323d3a96f712488221d1259640e1402728085 Mon Sep 17 00:00:00 2001
From: Thomas Voegtle <tv@lio96.de>
Date: Thu, 9 Jun 2022 15:19:22 +0200
Subject: [PATCH] xt_ECHO: use flowi6_to_flowi_common starting Linux 5.10.121

Upstream commit 3df98d79215a "lsm,selinux: pass flowi_common instead of flowi
to the LSM hooks" was backported to Linux 5.10.121 and you can't use flowi
anymore.

Lower check to 5.10.121.

Signed-off-by: Thomas Voegtle <tv@lio96.de>
---
 extensions/xt_ECHO.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extensions/xt_ECHO.c b/extensions/xt_ECHO.c
index 1ead749..cdd9238 100644
--- a/extensions/xt_ECHO.c
+++ b/extensions/xt_ECHO.c
@@ -101,7 +101,7 @@ echo_tg6(struct sk_buff *oldskb, const struct xt_action_param *par)
 	memcpy(&fl.daddr, &newip->daddr, sizeof(fl.daddr));
 	fl.fl6_sport = newudp->source;
 	fl.fl6_dport = newudp->dest;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 121)
 	security_skb_classify_flow((struct sk_buff *)oldskb, flowi6_to_flowi_common(&fl));
 #else
 	security_skb_classify_flow((struct sk_buff *)oldskb, flowi6_to_flowi(&fl));
-- 
2.25.1

